kernel/byterun/coq_interp.c

```diff
@@ -422,6 +429,17 @@ value coq_interprete
       }
       Next;
 
+      Instruct(ENSURESTACKCAPACITY) {
+        print_instr("ENSURESTACKCAPACITY");
+        int size = *pc++;
+        if (sp - size < coq_stack_low) {
+          coq_sp = sp;
+          realloc_coq_stack(size * sizeof(value));
+          sp = coq_sp;
+        }
```

Comment:
This seems incorrect. We might end up with no redzone once the stack has been extended and filled, if we are unlucky. Why is this code not just `CHECK_STACK(size)` so as to ensure the redzone is accounted for?